src/NodaTime/Calendars/WondrousYearMonthDayCalculator.cs

```diff
@@ -0,0 +1,385 @@
+
+            ConsoleWriteLine("~B GetDayFromStartOfYearToStartOfMonth {0} {1} ==> {2}", year, month,
+                daysFromStartOfYearToStartOfMonth);
+            return daysFromStartOfYearToStartOfMonth;
+        }
+
+        /// <param name="wYear"></param>
+        /// <remarks>Rely on Gregorian calendar for "days" in unix epoch</remarks>
+        /// <remarks>Untested</remarks>
+        /// <returns></returns>
+        protected override int CalculateStartOfYearDays(int wYear)
+        {
+            Preconditions.CheckArgumentRange(nameof(wYear), wYear, WondrousMinYear, WondrousMaxYear);
+
+            var gregorianYear = wYear + 1843;
+            var gregorianStart = new GregorianYearMonthDayCalculator().GetStartOfYearInDays(gregorianYear);
+
+            var nawRuz = GregorianDateOfNawRuz(wYear);
+
+            var firstDay = gregorianStart + nawRuz.DayOfYear - 1;
+
+            ConsoleWriteLine("~C CalcStartOfYearDays {0} ==> {1} ==> {2}", wYear, nawRuz, firstDay);
+
+            return firstDay;
+        }
+
+        /// <summary>
+        /// Test this possible date
+        /// </summary>
+        /// <param name="year"></param>
+        /// <param name="month">Ayyam-i-Ha is appended to month 18</param>
+        /// <param name="day"></param>
+        internal override void ValidateYearMonthDay(int year, int month, int day)
+        {
+            ConsoleWriteLine("~D Validate: {0}-{1}-{2}", year, month, day);
+
+            Preconditions.CheckArgumentRange(nameof(year), year, WondrousMinYear, WondrousMaxYear);
+
+            if (month == Month18)
+            {
+                // allow Ayyam-i-Ha days in this month
+                Preconditions.CheckArgumentRange(nameof(day), day, 1, DaysInMonth + DaysInAyyamHa(year));
+            }
+            else
+            {
+                Preconditions.CheckArgumentRange(nameof(month), month, 1, MonthsInYear);
+                Preconditions.CheckArgumentRange(nameof(day), day, 1, DaysInMonth);
+            }
+        }
+
+        /// <param name="year"></param>
+        /// <remarks>All years have 19 months</remarks>
+        /// <remarks>Tested</remarks>
+        /// <returns></returns>
+        internal override int GetMonthsInYear(int year)
+        {
+            return MonthsInYear;
+        }
+
+        /// <param name="year"></param>
+        /// <param name="month"></param>
+        /// <remarks>All months have 19 days. Ayyam-i-Ha usually has 4 or 5 days.</remarks>
+        /// <returns></returns>
+        internal override int GetDaysInMonth(int year, int month)
+        {
+            Preconditions.CheckArgumentRange(nameof(year), year, WondrousMinYear, WondrousMaxYear);
+
+            var daysInMonth = DaysInMonth;
+
+            if (month == AyyamiHaMonth0)
+            {
+                daysInMonth = WondrousCalendarHelper.GetYearInfo(year).DaysInAyyamiHa;
+            }
+
+            ConsoleWriteLine("~E GetDaysInMonth {0} {1} ==> {2}", year, month, daysInMonth);
```

Comment:
Please remove all of these diagnostic calls.

Reply:
Agreed.